src/design-system/components/Text/Text.tsx

```diff
@@ -183,58 +203,71 @@ const RawText = memo(
-  >
+  [key in T]: React.ForwardRefExoticComponent<PropsWithoutRef<TextProps> & RefAttributes<BaseText>>
 }
 
-const Text = generateStyledComponent(RawText) as TextComposition
+const Text = generateStyledComponent(RawText) as TextComposition<TextVariant>
+
 const generateTextVariant = (variant: TextVariant) =>
-  forwardRef<BaseText, TextProps>((props, ref) => <Text variant={variant} {...props} ref={ref} />)
-
-Text.H1 = generateTextVariant('H1')
-Text.H1Bold = generateTextVariant('H1Bold')
-Text.H2 = generateTextVariant('H2')
-Text.H2Bold = generateTextVariant('H2Bold')
-Text.H3 = generateTextVariant('H3')
-Text.H3Bold = generateTextVariant('H3Bold')
-Text.H4 = generateTextVariant('H4')
-Text.H4Bold = generateTextVariant('H4Bold')
-Text.H5 = generateTextVariant('H5')
-Text.H5Bold = generateTextVariant('H5Bold')
-Text.H6 = generateTextVariant('H6')
-Text.H6Bold = generateTextVariant('H6Bold')
-Text.Body = generateTextVariant('Body')
-Text.BodyBold = generateTextVariant('BodyBold')
-Text.Caption = generateTextVariant('Caption')
-Text.CaptionBold = generateTextVariant('CaptionBold')
-Text.Subtitle = generateTextVariant('Subtitle')
-Text.SubtitleBold = generateTextVariant('SubtitleBold')
-Text.NavLabel = generateTextVariant('NavLabel')
-Text.NavLabelBold = generateTextVariant('NavLabelBold')
-
-export { Text }
+  forwardRef<BaseText, TextProps>((props, ref) => <Text {...{ ...props, ref, variant }} />)
+
+Text.LgBold = generateTextVariant('LgBold')
+Text.LgMedium = generateTextVariant('LgMedium')
+Text.LgRegular = generateTextVariant('LgRegular')
+Text.LgSemibold = generateTextVariant('LgSemibold')
+Text.MdBold = generateTextVariant('MdBold')
+Text.MdMedium = generateTextVariant('MdMedium')
+Text.MdRegular = generateTextVariant('MdRegular')
+Text.MdSemibold = generateTextVariant('MdSemibold')
+Text.SmBold = generateTextVariant('SmBold')
+Text.SmMedium = generateTextVariant('SmMedium')
+Text.SmRegular = generateTextVariant('SmRegular')
+Text.SmSemibold = generateTextVariant('SmSemibold')
+Text.XsBold = generateTextVariant('XsBold')
+Text.XsMedium = generateTextVariant('XsMedium')
+Text.XsRegular = generateTextVariant('XsRegular')
+Text.XsSemibold = generateTextVariant('XsSemibold')
+Text.XlBold = generateTextVariant('XlBold')
+Text.XlMedium = generateTextVariant('XlMedium')
+Text.XlRegular = generateTextVariant('XlRegular')
+Text.XlSemibold = generateTextVariant('XlSemibold')
+
+export { Display, Text }
+
+const Display = generateStyledComponent(RawText) as TextComposition<DisplayVariant>
+
+const generateDisplayVariant = (variant: DisplayVariant = 'MdRegular') =>
+  forwardRef<BaseText, TextProps>((props, ref) => {
+    return <Display {...{ ...props, ref, variant }} type="display" />
+  })
+
+Display['2xlBold'] = generateDisplayVariant('2xlBold')
+Display['2xlMedium'] = generateDisplayVariant('2xlMedium')
+Display['2xlRegular'] = generateDisplayVariant('2xlRegular')
+Display['2xlSemibold'] = generateDisplayVariant('2xlSemibold')
```

Comment:
Maybe we could ask designers to change 2xl to xxl 🤔

Reply:
Good idea! I will ask

Reply:
Maybe, even if it could not be achieved, we could use this convention in codebase 🤔